cmd/process-user-data/main.go

```diff
@@ -0,0 +1,354 @@
+	if err != nil {
+		return fmt.Errorf("failed to write userData to file: %s", err)
+	}
+
+	fmt.Printf("Wrote userData to file: %s\n", dstFilePath)
+
+	return nil
+
+}
+
+// Get daemon.Config from userData
+func getConfigFromUserData(userData string) daemon.Config {
+
+	// UnMarshal the userData into forwarder (daemon) Config struct
+	var daemonConfig daemon.Config
+
+	err := json.Unmarshal([]byte(userData), &daemonConfig)
+	if err != nil {
+		fmt.Printf("failed to unmarshal userData: %s\n", err)
+		return daemon.Config{}
+	}
+
+	return daemonConfig
+}
+
+// Add method to get the value of aa-kbc-param from userdata and replace the value of aa_kbc_params in the
+// /etc/agent-config.toml file
+func updateAAKBCParams(aaKBCParams string, agentConfigFile string) error {
+
+	// if aaKBCParams is empty then return
+	if aaKBCParams == "" || agentConfigFile == "" {
+		return fmt.Errorf("aaKBCParams or agentConfigFile is empty")
+	}
+	// Replace the aa_kbc_params line in agentConfigFile with the aaKBCParams value
+	// Read the agentConfigFile
+	agentConfig, err := os.ReadFile(agentConfigFile)
+	if err != nil {
+		return fmt.Errorf("failed to read agent config file: %s", err)
+	}
+
+	// Split the agentConfigFile into lines
+	lines := strings.Split(string(agentConfig), "\n")
+
+	// Loop through the lines and replace the line that starts with aa_kbc_params
+	for i, line := range lines {
+		if strings.Contains(line, "aa_kbc_params") {
+			lines[i] = fmt.Sprintf("aa_kbc_params = \"%s\"", aaKBCParams)
+			fmt.Printf("Updated line: %s\n", lines[i])
+		}
+	}
+
+	// Join the lines back into a string
+	newAgentConfig := strings.Join(lines, "\n")
+
+	// Write the newAgentConfig to the agentConfigFile
+	err = os.WriteFile(agentConfigFile, []byte(newAgentConfig), 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write agent config file: %s", err)
+	}
+
+	fmt.Printf("Updated agent config file: %s\n", agentConfigFile)
+
+	return nil
+}
+
+func (cfg *Config) Setup() (cmd.Starter, error) {
+
+	var (
+		showVersion          bool
+		userDataFetchTimeout time.Duration
+		userData             string
+		ctx                  context.Context
+		cancel               context.CancelFunc
+		disableCloudConfig   bool
+	)
+
+	cmd.Parse(programName, os.Args, func(flags *flag.FlagSet) {
+		flags.BoolVar(&showVersion, "version", false, "Show version")
+		flags.StringVar(&cfg.configPath, "config", daemon.DefaultConfigPath, "Path to a deamon config file")
+		// flag to specify whether to use the VM's userData to configure the agent
+		flags.BoolVar(&disableCloudConfig, "disable-cloud-config", false, "Disable cloud config and use userdata via metadata servicee")
+		// flag to specify the timeout for retrieving the VM's userData
+		flags.DurationVar(&userDataFetchTimeout, "userdata-fetch-timeout", 180, "Timeout (in seconds) for retrieving the VM's userData. Default is 180s.")
+
+	})
+
+	cmd.ShowVersion(programName)
+
+	if showVersion {
+		cmd.Exit(0)
+	}
+
+	if disableCloudConfig {
```

Comment:
I assume we have this noop flag here because we cannot tell at runtime whether we need to run he tool? I think we can define that at buildtime (see the comment on systemd overrides)